tests/test_fields.py

```diff
@@ -0,0 +1,258 @@
+import sys
+import os
+
+# Sort the paths out to run from this file
+parent_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+src_path = os.path.join(os.path.sep,parent_path, 'src')
+sys.path.append(src_path)
+
+from fields import ScalarField, VectorField
+from regressors import CSModel, LModel
+import numpy as np
+import unittest
+
+
+# Key
+# g means grid test (2D field plane)
+# l means line test (1D field line)
```

Comment:
It is better to simply write grid and line in the tests they appear in rather than use this key. This only adds 3 characters so will not cause the code to appear bloated and will improve readability. Code should read well both in and out of order, so don't assume the reader of a function has necessarily read the comments at the top of the file. In particular, object and function names should be a stand-alone and self-explanatory as you can manage.

A good rule of thumb is to remember that code is read more often than it is written. If something takes longer to write but improves readability, it is worth doing.